docker/Dockerfile.sycl

```diff
@@ -119,8 +119,8 @@ RUN SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
     rm -rf ${SCRATCH_DIR}
 
 # Install Kokkos
-ARG KOKKOS_VERSION=4.1.00
-ARG KOKKOS_OPTIONS="-DKokkos_ENABLE_SYCL=ON -DCMAKE_CXX_FLAGS=-Wno-unknown-cuda-version -DKokkos_ENABLE_UNSUPPORTED_ARCHS=ON -DKokkos_ENABLE_DEPRECATED_CODE_3=OFF -DKokkos_ARCH_VOLTA70=ON -DCMAKE_CXX_STANDARD=17 -DCMAKE_CXX_FLAGS=-w"
+ARG KOKKOS_VERSION=4.3.00
+ARG KOKKOS_OPTIONS="-DKokkos_ENABLE_SYCL=ON -DCMAKE_CXX_FLAGS=-Wno-unknown-cuda-version -DKokkos_ENABLE_UNSUPPORTED_ARCHS=ON -DKokkos_ENABLE_DEPRECATED_CODE_3=OFF -DKokkos_ARCH_VOLTA70=ON -DKOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED=0 -DCMAKE_CXX_STANDARD=17 -DCMAKE_CXX_FLAGS=-w"
```

Comment:
@masterleinad What's the deal with that extra option?  Does Kokkos internal logic to set that variable fail?

Reply:
It's working fine for me on `polaris` and `A100`, i.e., newer versions of the compiler. We also explicitly disabled global device variable support in `Kokkos` since we had some problems, see https://github.com/kokkos/kokkos/blob/d88e2a5b0e7948c5315a605f5bc1ead00fc39fc6/.jenkins#L116.

Reply:
Also, this worked fine for Andrey for most test cases. We didn't quite understand why it was hanging in the `Kokkos` initialization for one if them. Again, this worked fine on `polaris`.